### PR TITLE
[WIP] feature/parsedown

### LINKED ISSUE #156: Add Github Flavored Markdown Support
Currently code blocks that are presented using the github flavored markdown will not be rendered correctly.

```markdown
    ```php
    $this->foo(); 
    $this->bar();
    ```
```
will result into ```$this->foo();$this->bar();``` (in lovely red color)

Considering all Modules are hosted on Github, we should support the GH flavored Markdown syntax at least for the codeblocks.

A lovely example how a description should **not** look like: http://modules.zendframework.com/manuakasam/SamVersioning

### LINKED ISSUE #169: Markdown in readme - code blocks errors.
Please compare the readme in one of these:

 + http://modules.zendframework.com/websafe/zf-mod-zfc-user-i18n-de-de
 + http://modules.zendframework.com/websafe/zf-mod-zfc-user-i18n-pl-pl
 + http://modules.zendframework.com/websafe/zf-mod-zfc-user-i18n-ja-JP

with the same files parsed by GitHub:

 + https://github.com/websafe/zf-mod-zfc-user-i18n-de-de/blob/master/README.md
 + https://github.com/websafe/zf-mod-zfc-user-i18n-pl-pl/blob/master/README.md
 + https://github.com/websafe/zf-mod-zfc-user-i18n-ja-JP/blob/master/README.md

The markdown parser on **modules.zendframework.com** has problems with code blocks enclosed in `~~~~`:
    ~~~~ php
    echo '123';
    ~~~~

Locally I'm testing my Markdown using https://github.com/michelf/php-markdown and the result looks pretty like GitHub flavored markdown.

### PR DESCRIPTION
**Pull request by ins0**
github markdown support

fixes #169 
fixes #156

### CODE REVIEW

File: module/Application/src/Application/Service/RepositoryRetriever.php [hunk 1]
```diff
@@ -69,7 +69,33 @@ public function getRepositoryFileContent($user, $module, $filePath)
...
+     */
+    private function requestContentMarkdown($content)
+    {
+        if (!$content) {
+            return $content;
+        }
+
+        try {
+            $content = $this->githubClient->api('markdown')->render($content);
+        } catch (RuntimeException $e) {
```
Comment (Ocramius) [inline module/Application/src/Application/Service/RepositoryRetriever.php]: Uhhh.....

Comment (Ocramius) [inline module/Application/src/Application/Service/RepositoryRetriever.php]: Should at least return `null` or such: rendering failed, and the consumer should simply skip echoing if `$rendered === null`

File: module/Application/src/Application/Service/RepositoryRetriever.php [hunk 2]
```diff
@@ -69,7 +69,33 @@ public function getRepositoryFileContent($user, $module, $filePath)
...
+     * Request content as parsed markdown
+     *
+     * @param string $content
+     *
+     * @return string
+     */
+    private function requestContentMarkdown($content)
+    {
+        if (!$content) {
+            return $content;
```
Comment (Ocramius) [inline module/Application/src/Application/Service/RepositoryRetriever.php]: Also weird

File: composer.json [hunk 3]
```diff
@@ -9,7 +9,7 @@
         "zendframework/zendframework": "~2.3.0",
         "rwoverdijk/assetmanager": "1.3.*",
         "evandotpro/edp-github": "dev-master",
-        "evandotpro/edp-markdown": "0.2.*",
+        "erusev/parsedown": "1.5.*",
```
Comment (Ocramius) [inline composer.json]: Unused?

File: module/Application/src/Application/Service/RepositoryRetriever.php [hunk 4]
```diff
@@ -69,7 +69,33 @@ public function getRepositoryFileContent($user, $module, $filePath)
...
+        }
+
+        try {
+            $content = $this->githubClient->api('markdown')->render($content);
```
Comment (Ocramius) [inline module/Application/src/Application/Service/RepositoryRetriever.php]: Can we trust this?

Comment (ins0) [inline module/Application/src/Application/Service/RepositoryRetriever.php]: i think more than a parser lib and full markdown support as github, security comes in combination with the htmlpurifier cleaner

File: composer.json (hunk 3 above)
Comment (ins0) [inline composer.json]: i tested a few markdown libs but failed rendering "special" user readme files. will be removed

File: module/Application/src/Application/Service/RepositoryRetriever.php (hunk 1 above)
Comment (ins0) [inline module/Application/src/Application/Service/RepositoryRetriever.php]: yep had the same felling as i pushed this pr, wasn't sure what this method should be return on errors so thanks for the suggestion :)

File: module/Application/src/Application/Service/RepositoryRetriever.php (hunk 2 above)
Comment (ins0) [inline module/Application/src/Application/Service/RepositoryRetriever.php]: sign

File: module/Application/src/Application/Service/RepositoryRetriever.php [hunk 5]
```diff
@@ -69,7 +69,33 @@ public function getRepositoryFileContent($user, $module, $filePath)
             return false;
         }
 
-        return base64_decode($contentResponse->content);
+        $content = base64_decode($contentResponse->content);
+        if ($content && $parseMarkdown) {
+            return $this->requestContentMarkdown($content);
```
Comment (ins0) [inline module/Application/src/Application/Service/RepositoryRetriever.php]: @Ocramius should this return the unparsed content when the rendering failed? i would say yes -no?

Comment (Ocramius) [inline module/Application/src/Application/Service/RepositoryRetriever.php]: No, it should probably just return `null`

**Comment by GeeH** [comment]
Can this be merged @Ocramius?

**Comment by gianarb** [comment]
Beautiful improvement! :) Thanks @ins0 for your work!

**Comment by Ocramius** [comment]
Looks good but I won't merge while on the phone :-)

**Comment by Ocramius** [comment]
Le merge! Thanks @ins0!

**Pull request merged**